android/src/com/mapswithme/maps/bookmarks/BookmarksListFragment.java

```diff
@@ -739,26 +712,95 @@ public boolean onOptionsItemSelected(MenuItem item)
 
     if (item.getItemId() == R.id.bookmarks_more)
     {
-      BottomSheet bs = BottomSheetHelper.create(requireActivity(),
-          mCategoryDataSource.getData().getName())
-          .sheet(R.menu.menu_bookmarks_list)
-          .listener(this::onListMoreMenuItemClick)
-          .build();
-
-      @BookmarkManager.SortingType int[] types = getAvailableSortingTypes();
-      Menu moreMenu = bs.getMenu();
-      moreMenu.findItem(R.id.sort).setVisible(types.length > 0 && !isEmpty());
-      moreMenu.findItem(R.id.delete_category).setVisible(!isLastOwnedCategory());
-      moreMenu.findItem(R.id.share_category).setVisible(!isEmpty());
-
-      BottomSheetHelper.tint(bs);
-      bs.show();
+      new MenuBottomSheetFragment.Builder()
+              .setTitle(mCategoryDataSource.getData().getName())
+              .setMenuBottomSheetItems(getOptionsMenuItems())
+              .show(getParentFragmentManager(), "optionsBottomSheet");
       return true;
     }
 
     return super.onOptionsItemSelected(item);
   }
 
+  private void onOptionsMenuItemClicked(MenuBottomSheetItem item)
+  {
+    switch (item.getId())
+    {
+      case BottomSheetItemId.SORT:
+        ChooseBookmarksSortingTypeFragment.chooseSortingType(getAvailableSortingTypes(),
```

Comment:
Again, is it possible to call these functions from the functor, passed in the ArrayList of menu items? Then all these click wrappers and id enums will be unnecessary.